common/utils/data_generator/data_utils.py

```diff
@@ -0,0 +1,65 @@
+import json
+import random
+
+from shapely.geometry import Point, Polygon
+
+poly = [
+    (37.75850848099701, -122.50833008408812),
+    (37.75911919711413, -122.49648544907835),
+    (37.751620611284935, -122.4937388670471),
+    (37.74863453749236, -122.50742886185911),
+]
+
+
+def polygon_random_points(polygon: list, num_points: int) -> list:
+    """
+    This function will generate points by a given polygon
+    :param polygon: Polygon coordinates of the desired points zone
+    :param num_points: points amount to be generated
+    :return:
+    list of lat long values
+    """
+    polygon = Polygon(polygon)
+    min_x, min_y, max_x, max_y = polygon.bounds
+    points = []
+    while len(points) < num_points:
+        random_point = Point(
+            [random.uniform(min_x, max_x), random.uniform(min_y, max_y)]
+        )
+        if random_point.within(polygon):
+            point_val = (float(random_point.x), float(random_point.y))
+            points.append(point_val)
+    return points
+
+
+def generate_points_request(
```

Comment:
points_amount: int, polygon: List[Tuple[float, float]], exclude_fields: bool, product_type: str = "MIXED"

Make it harder - double check me

Reply:
```
def generate_points_request(
    points_amount: int, polygon: List[Tuple[float, float]], exclude_fields: bool, product_type: str = "MIXED"
) 
```